### ISSUE TITLE
Unexpected error: Unused variable '__class__'

### ISSUE THREAD

**Issue by paulsef** [PyCQA/pylint#1790]
### Steps to reproduce
The following snippet will fail with the versions described below:
1. A file named `example.py` with the following snippet:
```python
class Class:
    @classmethod
    def _method__(cls):
        cls.__subclasses__()
```
2. `pylint example.py`

### Current behavior
The following error is the result of the above:
`W:  1, 0: Unused variable '__class__' (unused-variable)`

Note, this problem appears to be related to the astroid version. Pinning the astroid version to 1.5.3 solves the problem.

### Expected behavior
This should pass the linter.

### pylint --version output
```
pylint 1.7.2, 
astroid 1.6.0
Python 3.6.2 (default, Sep 14 2017, 09:50:24) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

**Comment by PCManticore** [comment]
you'll need to update to 1.7.5 or 1.8 to have this fixed.
